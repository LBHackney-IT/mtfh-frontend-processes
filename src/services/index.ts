export { default as locale } from "./locale";
export { default as menu } from "./menu";
export * as processes from "./processes";
export * from "./config";
