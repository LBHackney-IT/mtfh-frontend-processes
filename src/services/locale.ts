import reactHtmlParser from "react-html-parser";

import { AssetAddress } from "@mtfh/common/lib/api/asset/v1";

const locale = {
  backButton: "Back",
  bookAppointment: "Book Appointment",
  next: "Next",
  cancel: "Cancel",
  cancelProcess: "Cancel process",
  change: "Change",
  views: {
    soleToJoint: {
      title: "Sole tenant requests a joint tenure",
      steps: {
        selectTenant: "Select person",
        checkEligibility: "Eligibility Checks",
        supportingDocuments: "Supporting documents",
        finish: "Finish",
        breachOfTenancy: "Breach of tenure check",
        requestDocuments: "Request Documents",
        reviewDocuments: "Review Documents",
        submitCase: "Submit case",
      },
      actions: {
        reassignCase: "Reassign case",
        cancelProcess: "Cancel process",
        caseActivityHistory: "Case activity history",
      },
    },
    selectTenants: {
      selectTenantHint:
        "This person will be asked for proof of relationship e.g. marriage or civil partnership certificate.",
      selectTenantLabel: "Who do you want to add as a joint tenant?",
      addToTenureText: "If the person you want to add is not listed you must first",
      addToTenureLink: "add them to the tenure",
      noHouseholdMembersOver18:
        "There are no household members over the age of 18 associated to this.",
    },
    processesMenu: {
      selectProcessDescription: () =>
        reactHtmlParser(
          "<p>Start a process by selecting a process category below. <br /> Where a process is not yet supported in <strong>Manage My Home</strong> these links will open the appropriate interim Google form.</p>",
        ),
    },
    checkEligibility: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Passed automatic eligibilty checks",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    breachOfTenancy: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Eligibility checks passed",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    reviewDocuments: {
      passedChecks: "Eligibility checks passed",
      documentsRequested: "Documents requested",
      checkSupportingDocumentsAppointment:
        "I need to make an appointment to check supporting documents",
      appointmentScheduled: "Office appointment scheduled",
      seenPhotographicId:
        "I confirm that an outcome letter has been sent to the resident",
      seenSecondId:
        "I confirm I have seen a second form of ID (does not have to be photographic)",
      isNotInImmigrationControl:
        "I confirm that the prospective tenant is not subject to immigration control under the Asylum And Immigration Act 1996",
      seenProofOfRelationship:
        "I confirm I have seen proof of relationship to the existing tenant",
      incomingTenantLivingInProperty:
        "I confirm I have seen 3 separate documents proving the proposed tenant has been living at the property for a minimum of 12 months",
    },
  },
  errors: {
    unableToFetchRecord: "There was a problem retrieving the record",
    unableToFetchRecordDescription:
      "Please try again. If the issue persists, please contact support.",
    unableToFindProcess: "Invalid Process",
    unableToFindProcessDescription: "We have been unable to find a process of this type.",
    unableToFindState: "Invalid State",
    unableToFindStateDescription: "We have been unable to process a state of this type.",
  },
  title: "Processes",
  loadingText: "Submitting...",
  components: {
    startProcess: {
      buttonLabel: "Start process",
      thirdPartyHeading: "Sharing information with third parties",
    },
    entitySummary: {
      tenurePaymentRef: "Tenure payment ref",
      address: (assetAddress: AssetAddress): string => {
        const {
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,
          addressLine4,
          postCode,
        } = assetAddress;

        return [
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,

          addressLine4,
          postCode,
        ]
          .filter((addressLine) => !!addressLine)
          .join(" ");
      },
    },
  },
};

export default locale;
