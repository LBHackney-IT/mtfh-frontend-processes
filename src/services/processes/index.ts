export * from "./sole-to-joint";
export * from "./change-of-name";
