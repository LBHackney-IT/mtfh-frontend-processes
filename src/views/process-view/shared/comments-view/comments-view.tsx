import React, { useState } from "react";
import { Link as RouterLink } from "react-router-dom";

import { Form, Formik } from "formik";

import { CommentsFormData } from "../../../../schemas/comments";
import { locale } from "../../../../services";

import { addComment } from "@mtfh/common/lib/api/comments/v2";
import {
  Button,
  Center,
  CommentList,
  Dialog,
  DialogActions,
  ErrorSummary,
  Field,
  Heading,
  Input,
  Link,
  Spinner,
  TextArea,
} from "@mtfh/common/lib/components";
import { useErrorCodes } from "@mtfh/common/lib/hooks";

import "../../sole-to-joint-view/styles.scss";

export const CommentsView = ({ targetType, targetId, mutate }) => {
  const [addCommentOpen, setAddCommentOpen] = useState<boolean>(false);
  const [closeCommentOpen, setCloseCommentOpen] = useState<boolean>(false);
  const [refreshTargetId, setRefreshTargetId] = useState<string | undefined>();
  const [error, setError] = useState<string | undefined>();
  const errorMessages = useErrorCodes();

  if (!errorMessages) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const { W1: correctIndicatedErrorsText } = errorMessages;
  return (
    <>
      <Heading variant="h4" style={{ marginBottom: "0.5em" }}>
        Comments
      </Heading>

      {addCommentOpen ? (
        <>
          <Formik<CommentsFormData>
            initialValues={{ description: "", title: "" }}
            onSubmit={async (values, { setErrors }) => {
              const { description, title } = values;
              setError(undefined);
              try {
                await addComment({
                  description,
                  title,
                  targetType,
                  targetId,
                  highlight: false,
                });
                setAddCommentOpen(false);
                mutate();
              } catch (error: any) {
                if (typeof error === "object" && error.isAxiosError === true) {
                  const errors: Record<string, string> = {};
                  if (error.response.data.errors) {
                    Object.entries(error.response.data.errors).forEach(([key, value]) => {
                      errors[key.charAt(0).toLowerCase() + key.slice(1)] = JSON.parse(
                        value as string,
                      ).errorMessage;
                    });
                  }
                  switch (error.response.status) {
                    case 400:
                    case 404:
                      setErrors(errors);
                      return setError("invalid");
                    default:
                      return setError("error");
                  }
                }
              } finally {
                setRefreshTargetId("a-temp-id"); // to force CommentList refresh itself
                setRefreshTargetId(undefined);
              }
            }}
          >
            {(props) => {
              const {
                handleChange,
                handleBlur,
                values: { description, title },
              } = props;
              const hasFieldErrors = Object.keys(props.errors).length > 0;
              return (
                <Form noValidate id="comment-form" className="comment-form">
                  {(hasFieldErrors || error) && (
                    <ErrorSummary
                      id="add-comments-error"
                      title={locale.errors.errorLabel}
                    >
                      <>
                        {hasFieldErrors && <p>{correctIndicatedErrorsText}</p>}
                        {error && <p>{locale.errors.somethingWentWrongLabel}</p>}
                      </>
                    </ErrorSummary>
                  )}

                  <Field id="comment-title" name="title" label="Comment title" required>
                    <Input data-testid="comment-form:title" />
                  </Field>
                  <Field
                    id="comment-description"
                    name="description"
                    label="Comment"
                    required
                  >
                    <TextArea
                      rows={10}
                      maxLength={1000}
                      onChange={handleChange}
                      onBlur={handleBlur}
                      value={description}
                      data-testid="comment-form:description"
                    />
                  </Field>
                  <Button
                    type="submit"
                    data-testid="comment-form-submit"
                    disabled={!description || !title}
                  >
                    Save comment
                  </Button>
                </Form>
              );
            }}
          </Formik>

          <Dialog
            isOpen={closeCommentOpen}
            onDismiss={() => setCloseCommentOpen(false)}
            title="Do you want to cancel this comment? All progress will be lost"
          >
            <DialogActions className="closeCommentDialogActions">
              <Button
                data-testid="close-comment-modal-yes"
                onClick={() => {
                  setCloseCommentOpen(false);
                  setAddCommentOpen(false);
                }}
              >
                Yes
              </Button>
              <Button variant="secondary" onClick={() => setCloseCommentOpen(false)}>
                No
              </Button>
            </DialogActions>
          </Dialog>

          <Link
            as={RouterLink}
            to="#"
            variant="link"
            onClick={() => setCloseCommentOpen(true)}
          >
            Cancel comment
          </Link>
        </>
      ) : (
        <>
          <Button
            style={{ marginTop: "1em" }}
            data-testid="add-comment"
            variant="secondary"
            onClick={() => {
              setAddCommentOpen(true);
            }}
          >
            Add comment
          </Button>
          <div>
            <CommentList targetId={refreshTargetId || targetId} />
          </div>
        </>
      )}
    </>
  );
};
