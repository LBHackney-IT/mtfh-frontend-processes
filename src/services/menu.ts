import * as processes from "./processes";

import { Asset } from "@mtfh/common/lib/api/asset/v1/types";
import { Tenure } from "@mtfh/common/lib/api/tenure/v1/types";

interface MenuProps {
  label: string;
  link: string;
  processes?: { label: string; link: string; targetType: string }[];
  getPplQuery?: (context: any) => string;
}

const getProcessLink = (processName: string) => {
  const process = processes[processName];
  return {
    targetType: process.targetType,
    label: process.title,
    link: `/processes/${process.processName}/start`,
  };
};

const getPplQueryForRequestALegalReferral = ({ tenure, asset }): string => {
  if (!tenure || !asset) {
    return "";
  }
  const {
    householdMembers,
    tenuredAsset: { propertyReference },
  } = tenure as Tenure;
  const {
    assetAddress: { addressLine1, postCode },
  } = asset as Asset;
  const tenantNames = householdMembers
    .filter((member) => member.isResponsible)
    .map((member) => member.fullName)
    .join(", ");
  const address = addressLine1;
  const query = `usp=pp_url&entry.1699199605=${tenantNames}&entry.2086190845=${address}&entry.1956036523=${postCode}&entry.1140527432=${propertyReference}`;
  const encodedQuery = encodeURI(query);
  return encodedQuery;
};

const getPplQueryTenancyChange = ({ tenure }): string => {
  if (!tenure) {
    return "";
  }
  const {
    householdMembers,
    paymentReference,
    tenuredAsset: { fullAddress, propertyReference },
  } = tenure as Tenure;

  const tenant = householdMembers.find((member) => member.isResponsible);
  const query = `usp=pp_url&entry.1942863541=${tenant?.fullName}&entry.1090481808=${fullAddress}&entry.321634337=${propertyReference}&entry.582205322=${paymentReference}`;
  const encodedQuery = encodeURI(query);
  return encodedQuery;
};

const menu: MenuProps[] = [
  {
    label: "Sole to Joint",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform",
    processes: [
      getProcessLink(processes.soletojoint.processName),
      getProcessLink(processes.changeOfName.processName),
    ],
  },
  {
    label: "Other tenancy changes",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform",
    getPplQuery: getPplQueryTenancyChange,
  },
  {
    label: "Record a tenancy / audit homecheck",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform",
  },
  {
    label: "Find legal referral response",
    link: "https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140",
  },
  {
    label: "Request a legal referral",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform",
    getPplQuery: getPplQueryForRequestALegalReferral,
  },
  {
    label: "Record notice served",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform",
  },
  {
    label: "Rehousing",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform",
  },
  {
    label: "Person",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform",
  },
  {
    label: "Report Japanese knotweed",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628",
  },
  {
    label: "Report overgrown tree",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628",
  },
  {
    label: "Report overgrown garden",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628",
  },
];

export default menu;
