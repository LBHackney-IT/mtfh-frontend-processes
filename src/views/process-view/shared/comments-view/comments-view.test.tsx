import React from "react";

import { postCommentV2, render, server } from "@hackney/mtfh-test-utils";
import { screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { locale } from "../../../../services";
import { CommentsView } from "./comments-view";

import * as commentsV2 from "@mtfh/common/lib/api/comments/v2/services";
import { CommentsResponse } from "@mtfh/common/lib/api/comments/v2/services";
import * as referenceDataV1 from "@mtfh/common/lib/api/reference-data/v1/service";
import { AxiosSWRInfiniteResponse, AxiosSWRResponse } from "@mtfh/common/lib/http";

const targetType = "process";
const targetId = "process-id-0";
const mutate = () => undefined;

const fullName = "Name Surname";
const email = "email@test.com";

const commentsResponse: CommentsResponse[] = [
  {
    results: [
      {
        id: "fbf6e1be-4045-4d73-a93b-6c50539cfc9c",
        title: null,
        description: "test description 1",
        targetType: "tenure",
        targetId: "c59d1888-b288-0fff-706b-ff3df2cbd51e",
        createdAt: "2022-06-13T13:48:27.562Z",
        categorisation: {
          category: null,
          subCategory: null,
          description: null,
        },
        author: {
          fullName,
          email,
        },
        highlight: false,
      },
      {
        id: "da5a7f4c-ee1e-482d-a955-7977e5ad5daa",
        title: null,
        description: "test description 2",
        targetType: "tenure",
        targetId: "c59d1888-b288-0fff-706b-ff3df2cbd51e",
        createdAt: "2022-06-13T13:29:33.938Z",
        categorisation: {
          category: null,
          subCategory: null,
          description: null,
        },
        author: {
          fullName,
          email,
        },
        highlight: false,
      },
    ],
    paginationDetails: {
      nextToken: "nextToken",
    },
  } as CommentsResponse,
];

const referenceData = {
  category: "category-1",
};

beforeEach(() => {
  jest.spyOn(commentsV2, "useComments").mockReturnValue({
    data: commentsResponse,
    size: 1,
  } as unknown as AxiosSWRInfiniteResponse<CommentsResponse>);

  jest.spyOn(referenceDataV1, "useReferenceData").mockReturnValue({
    data: referenceData,
  } as unknown as AxiosSWRResponse<any>);
});

afterEach(() => {
  jest.restoreAllMocks();
});

describe("CommentsView", () => {
  test("it renders comments by default", async () => {
    const { container } = render(
      <CommentsView targetType={targetType} targetId={targetId} mutate={mutate} />,
    );
    await expect(screen.findByText("test description 1")).resolves.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("it renders inputs to add comments when user clicks Add Comment", async () => {
    server.use(postCommentV2("", 200));
    const { container } = render(
      <CommentsView targetType={targetType} targetId={targetId} mutate={mutate} />,
    );

    await userEvent.click(screen.getByTestId("add-comment"));

    await expect(screen.findByText("Comment title")).resolves.toBeInTheDocument();
    await expect(screen.getByTestId("comment-form-submit")).toBeDisabled();

    expect(container).toMatchSnapshot();

    await userEvent.type(screen.getByTestId("comment-form:title"), "test");
    await userEvent.type(screen.getByTestId("comment-form:description"), "test");

    await expect(screen.getByTestId("comment-form-submit")).not.toBeDisabled();
    await userEvent.click(screen.getByTestId("comment-form-submit"));

    await expect(screen.findByText("Comment title")).resolves.not.toBeInTheDocument();
  });

  test(`it opens a modal when user click "Cancel comment" link`, async () => {
    const { container } = render(
      <CommentsView targetType={targetType} targetId={targetId} mutate={mutate} />,
    );

    await userEvent.click(screen.getByTestId("add-comment"));
    await userEvent.click(screen.getByText("Cancel comment"));

    await expect(
      screen.findByText("Do you want to cancel this comment? All progress will be lost"),
    ).resolves.toBeInTheDocument();

    expect(container).toMatchSnapshot();

    await userEvent.click(screen.getByText("No"));
    await expect(screen.queryByText("No")).toBeNull();
    await expect(screen.getByTestId("comment-form:title")).toBeInTheDocument();

    await userEvent.click(screen.getByText("Cancel comment"));
    await expect(
      screen.findByText("Do you want to cancel this comment? All progress will be lost"),
    ).resolves.toBeInTheDocument();
    await userEvent.click(screen.getByText("Close"));
    await expect(screen.queryByText("No")).toBeNull();
  });

  test(`it opens a modal when user click "Cancel comment" link and closes on dismiss`, async () => {
    render(<CommentsView targetType={targetType} targetId={targetId} mutate={mutate} />);

    await userEvent.click(screen.getByTestId("add-comment"));
    await userEvent.click(screen.getByText("Cancel comment"));

    await expect(
      screen.findByText("Do you want to cancel this comment? All progress will be lost"),
    ).resolves.toBeInTheDocument();

    await userEvent.click(screen.getByText("Close"));
    await expect(
      screen.queryByText("Do you want to cancel this comment? All progress will be lost"),
    ).toBeNull();
  });

  test(`it opens a modal when user click "Cancel comment" link and closes a modal when user click "Yes"`, async () => {
    render(<CommentsView targetType={targetType} targetId={targetId} mutate={mutate} />);

    await userEvent.click(screen.getByTestId("add-comment"));
    await userEvent.click(screen.getByText("Cancel comment"));

    await expect(
      screen.findByText("Do you want to cancel this comment? All progress will be lost"),
    ).resolves.toBeInTheDocument();

    await userEvent.click(screen.getByTestId("close-comment-modal-yes"));
    await expect(screen.queryByTestId("close-comment-modal-yes")).toBeNull();
  });

  test("it shows an errors form response", async () => {
    server.use(
      postCommentV2(
        {
          errors: {
            Description:
              '{\n  "errorCode": "W00",\n  "errorMessage": "Description error"}',
          },
        },
        400,
      ),
    );

    render(<CommentsView targetType={targetType} targetId={targetId} mutate={mutate} />);
    await userEvent.click(screen.getByTestId("add-comment"));
    await userEvent.type(screen.getByTestId("comment-form:title"), "test");
    await userEvent.type(screen.getByTestId("comment-form:description"), "test");
    await userEvent.click(screen.getByTestId("comment-form-submit"));

    await waitFor(() => {
      expect(screen.queryByText("Description error")).toBeInTheDocument();
    });
  });

  test("it shows an error if unable to post", async () => {
    server.use(postCommentV2("error", 500));

    render(<CommentsView targetType={targetType} targetId={targetId} mutate={mutate} />);
    await userEvent.click(screen.getByTestId("add-comment"));
    await userEvent.type(screen.getByTestId("comment-form:title"), "test");
    await userEvent.type(screen.getByTestId("comment-form:description"), "test");
    await userEvent.click(screen.getByTestId("comment-form-submit"));

    await waitFor(() => {
      expect(
        screen.queryByText(locale.errors.somethingWentWrongLabel),
      ).toBeInTheDocument();
    });
  });
});
