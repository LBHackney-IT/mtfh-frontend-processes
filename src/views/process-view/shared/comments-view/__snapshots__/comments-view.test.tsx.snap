// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsView it opens a modal when user click "Cancel comment" link 1`] = `
<div
  aria-hidden="true"
>
  <h4
    class="lbh-heading-h4"
    style="margin-bottom: 0.5em;"
  >
    Comments
  </h4>
  <form
    action="#"
    class="comment-form"
    id="comment-form"
    novalidate=""
  >
    <div
      class="govuk-form-group lbh-form-group"
      id="comment-title"
    >
      <label
        class="govuk-label lbh-label"
        for="comment-title-field"
      >
        Comment title
        <sup
          aria-hidden="true"
        >
          *
        </sup>
      </label>
      <input
        class="govuk-input lbh-input"
        data-testid="comment-form:title"
        id="comment-title-field"
        name="title"
        required=""
        value=""
      />
    </div>
    <div
      class="govuk-character-count"
    >
      <div
        class="govuk-form-group lbh-form-group"
        id="comment-description"
      >
        <label
          class="govuk-label lbh-label"
          for="comment-description-field"
        >
          Comment
          <sup
            aria-hidden="true"
          >
            *
          </sup>
        </label>
        <textarea
          class="govuk-textarea lbh-textarea lbh-character-count"
          data-testid="comment-form:description"
          id="comment-description-field"
          name="description"
          required=""
          rows="10"
        />
        <span
          aria-live="polite"
          class="govuk-hint govuk-character-count__message"
        >
          You have 1000 characters remaining
        </span>
      </div>
    </div>
    <button
      class="govuk-button lbh-button"
      data-testid="comment-form-submit"
      disabled=""
      type="submit"
    >
      Save comment
    </button>
  </form>
  <a
    class="govuk-link lbh-link lbh-link--no-visited-state"
    href="/"
  >
    Cancel comment
  </a>
</div>
`;

exports[`CommentsView it renders comments by default 1`] = `
<div>
  <h4
    class="lbh-heading-h4"
    style="margin-bottom: 0.5em;"
  >
    Comments
  </h4>
  <button
    class="govuk-button lbh-button govuk-button--secondary lbh-button--secondary"
    data-testid="add-comment"
    style="margin-top: 1em;"
    type="button"
  >
    Add comment
  </button>
  <div>
    <div>
      <div
        class="comment"
      >
        <div
          class="comment__item"
        >
          <div
            class="comment__date-time"
          >
            13/06/2022
          </div>
          <div
            class="comment__date-time"
          >
            13:48:27
          </div>
        </div>
        <div
          class="comment__item --center"
        >
          test description 1
        </div>
        <div
          class="comment__item"
        >
          Name Surname
        </div>
      </div>
      <div
        class="comment"
      >
        <div
          class="comment__item"
        >
          <div
            class="comment__date-time"
          >
            13/06/2022
          </div>
          <div
            class="comment__date-time"
          >
            13:29:33
          </div>
        </div>
        <div
          class="comment__item --center"
        >
          test description 2
        </div>
        <div
          class="comment__item"
        >
          Name Surname
        </div>
      </div>
      <nav
        class="lbh-simple-pagination"
      >
        <a
          class="lbh-simple-pagination__link lbh-simple-pagination__link--next"
          rel="next"
        >
          Next
          <svg
            fill="none"
            height="19"
            viewBox="0 0 11 19"
            width="11"
          >
            <path
              d="M1 18L9 9.5L1 1"
              stroke-width="2"
            />
          </svg>
        </a>
      </nav>
    </div>
  </div>
</div>
`;

exports[`CommentsView it renders inputs to add comments when user clicks Add Comment 1`] = `
<div>
  <h4
    class="lbh-heading-h4"
    style="margin-bottom: 0.5em;"
  >
    Comments
  </h4>
  <form
    action="#"
    class="comment-form"
    id="comment-form"
    novalidate=""
  >
    <div
      class="govuk-form-group lbh-form-group"
      id="comment-title"
    >
      <label
        class="govuk-label lbh-label"
        for="comment-title-field"
      >
        Comment title
        <sup
          aria-hidden="true"
        >
          *
        </sup>
      </label>
      <input
        class="govuk-input lbh-input"
        data-testid="comment-form:title"
        id="comment-title-field"
        name="title"
        required=""
        value=""
      />
    </div>
    <div
      class="govuk-character-count"
    >
      <div
        class="govuk-form-group lbh-form-group"
        id="comment-description"
      >
        <label
          class="govuk-label lbh-label"
          for="comment-description-field"
        >
          Comment
          <sup
            aria-hidden="true"
          >
            *
          </sup>
        </label>
        <textarea
          class="govuk-textarea lbh-textarea lbh-character-count"
          data-testid="comment-form:description"
          id="comment-description-field"
          name="description"
          required=""
          rows="10"
        />
        <span
          aria-live="polite"
          class="govuk-hint govuk-character-count__message"
        >
          You have 1000 characters remaining
        </span>
      </div>
    </div>
    <button
      class="govuk-button lbh-button"
      data-testid="comment-form-submit"
      disabled=""
      type="submit"
    >
      Save comment
    </button>
  </form>
  <a
    class="govuk-link lbh-link lbh-link--no-visited-state"
    href="/"
  >
    Cancel comment
  </a>
</div>
`;
